cocos/editor-support/cocostudio/ActionTimeline/CCBoneNode.cpp

```diff
@@ -0,0 +1,598 @@
+/****************************************************************************
+Copyright (c) 2014 cocos2d-x.org
+
+http://www.cocos2d-x.org
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+****************************************************************************/
+
+#include "base/CCDirector.h"
+#include "renderer/CCRenderer.h"
+#include "renderer/ccGLStateCache.h"
+#include "renderer/CCGLProgram.h"
+#include "renderer/CCGLProgramState.h"
+
+#include "CCBoneNode.h"
+#include "CCSkeletonNode.h"
+
+using namespace cocos2d;
+
+NS_TIMELINE_BEGIN
+
+BoneNode::BoneNode()
+: _isRackShow(true)
```

Comment:
Not all member variables are initialized. 

```
   cocos2d::CustomCommand _customCommand;
+    cocos2d::BlendFunc     _blendFunc;
+
+    bool              _isRackShow;
+    cocos2d::Color4F  _rackColor;
+    int               _rackLength;
+    int               _rackWidth;
+
+    cocos2d::Vector<BoneNode*> _childBones;
+    cocos2d::Vector<SkinNode*> _boneSkins;
+    SkeletonNode*              _rootSkeleton;
+private:
+    cocos2d::Vec2          _squareVertices[4];
+    cocos2d::Color4F       _squareColors[4];
+    cocos2d::Vec3          _noMVPVertices[4];
+    CC_DISALLOW_COPY_AND_ASSIGN(BoneNode);
```

Primitive types must be initialed in the class initializer.